### DISCUSSION TITLE
HookWebpackError: Unterminated string in JSON at position 67 (line 1 column 68)     at makeWebpackError (C:\Downloads\project\node_modules\webpack\lib\HookWebpackError.js:48:9)

### DISCUSSION THREAD

**Discussion by Likhitha4131** [module-federation/module-federation-examples#4030] [Q&A] [answered]
I am facing this particular error for multiple projects 
my next.config.js-
`const NextFederationPlugin = require("@module-federation/nextjs-mf");
 
const nextConfig = {
  reactStrictMode: true,
  basePath: "/dc",
  env: {
    BASE_PATH: process.env.BASE_PATH
  },
  eslint: {
    ignoreDuringBuilds: true
  },
  typescript: {
    ignoreBuildErrors: true
  },
  webpack(config) {
    const nextWebConfig = { ...config };
    nextWebConfig.resolve.alias.canvas = false;
    nextWebConfig.plugins.push(
      new NextFederationPlugin({
        name: "dc",
        filename: "static/chunks/remoteEntry.js",
        exposes: {
          './canned-report': './src/components/CannedReport.tsx'
        },
        shared: {
          // react: {
          //   singleton: true,
          //   requiredVersion: '18.2.0',
          // },
          // 'react-dom': {
          //   singleton: true,
          //   requiredVersion: '18.2.0',
          // },
        },
        
      })
    );
 
    nextWebConfig.module.rules.push({
      test: /@vtaits\/use-lazy-ref/,
      use: {
        loader: 'babel-loader',
        options: {
          presets: ['next/babel'],
          plugins: [
            ['@babel/plugin-transform-modules-commonjs', { 'allowTopLevelThis': true }]
          ]
        }
      }
    });
 
    return nextWebConfig;
  },generateBuildId: async () => {
    return "dc";
  },
};
 
module.exports = nextConfig;`
My package.json
{
  "name": "dc",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "cross-env NEXT_PRIVATE_LOCAL_WEBPACK=true next dev -p 8001",
    "build": "next build",
    "start": "next start -p 8001"
  },
  "dependencies": {
    "@emotion/react": "^11.11.4",
    "@emotion/styled": "^11.11.0",
    "@mantine/core": "^7.7.1",
    "@mantine/dates": "^7.7.1",
    "@mantine/hooks": "^7.7.1",
    "@material-ui/icons": "^4.11.3",
    "@module-federation/nextjs-mf": "^8.1.7",
    "@mui/icons-material": "^5.15.15",
    "@mui/lab": "^5.0.0-alpha.170",
    "@mui/material": "^5.15.14",
    "@mui/x-data-grid": "^6.19.10",
    "@mui/x-date-pickers": "^7.1.0",
    "@mui/x-tree-view": "^7.3.0",
    "@types/react-chartjs-2": "^2.5.7",
    "axios": "^1.6.8",
    "chart.js": "^4.4.2",
    "chartjs": "^0.3.24",
    "dayjs": "^1.11.10",
    "moment": "^2.30.1",
    "next": "14.1.4",
    "next-auth": "^4.24.7",
    "react": "^18.2.0",
    "react-chartjs": "^1.2.0",
    "react-chartjs-2": "^5.2.0",
    "react-dom": "^18.2.0",
    "react-select": "^5.7.0",
    "react-select-async-paginate": "^0.7.3",
    "sass": "^1.72.0",
    "webpack-cli": "^5.1.4"
  },
  "devDependencies": {
    "@babel/core": "^7.24.5",
    "@babel/plugin-transform-modules-commonjs": "^7.24.1",
    "@babel/preset-env": "^7.24.5",
    "@babel/preset-react": "^7.24.1",
    "@babel/runtime": "^7.24.6",
    "@types/node": "^20",
    "@types/react": "^18",
    "@types/react-dom": "^18",
    "autoprefixer": "^10.0.1",
    "babel-loader": "^9.1.3",
    "cross-env": "^7.0.3",
    "eslint": "^8",
    "eslint-config-next": "14.1.4",
    "postcss": "^8",
    "tailwindcss": "^3.3.0",
    "typescript": "^5"
  }
}
I am using windows 10.This is happening with multiple projects .Please help with this one

**Comment by Likhitha4131** [comment, marked as answer]
Has upgraded my module-federation version to 8.2.0 and it statred working

**Comment by VityaSchel** [comment]
still getting this on @module-federation/nextjs-mf but now with Unterminated string in JSON at position 55
:(

**Reply by VityaSchel** [reply]
I even tried exactly this setup
https://github.com/module-federation/module-federation-examples/tree/master/nextjs-v14/home
still doesn't work

**Reply by VityaSchel** [reply]
That's some fucking next level bullshit I spent 3 hours on
It only happens when path to the directory you're running next.js in (and your project) has spaces

**Comment by VityaSchel** [comment]
https://github.com/module-federation/core/issues/2684
